lock-api/src/main/java/com/palantir/lock/client/AuthenticatingMultiClientConjureTimelockService.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import com.palantir.atlasdb.timelock.api.LeaderTimes;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import java.util.Set;
+
+public interface AuthenticatingMultiClientConjureTimelockService {
```

Comment:
I _think_ I understand why we need the impl, but not the interface - I feel like this is a very specific interface (where I'd rather there is a more generic interface and AMCCTS is one of the implementations). Regardless, this is also possibly a misleading name: this always passes a token that is "Bearer omitted", which while technically is on the theme of authentication, feels a bit odd to call this an authenticating service.

Reply:
I wouldn't name this in this way: users elsewhere in the AtlasDB client code shouldn't need to know that authentication (or the lack thereof!) is a thing that's happening beyond this layer, which is an implementation detail. Conceptually this interface should be something like `InternalMultiClientConjureTimelockService` (possibly not a great name) and then your `AuthMCCTSImpl` should be `AuthIMCCTS` (or even `UnauthedIMCCTS`).

Reply:
I hit a mental block in trying to come up with a name for this interface, ```InternalMultiClientConjureTimelockService``` is reasonable.